src/distances.jl

```diff
@@ -34,3 +26,23 @@ with `T` of appropriate type for members of the set.
 function distance_to_set end
 
 distance_to_set(::AbstractDistance, v, s) = distance_to_set(DefaultDistance(), v, s)
+
+"""
+    EpigraphViolationDistance
+
+Distance used when `v ∈ s` can be expressed as a single scalar function `f(v) ≤ 0`.
+The distance is expressed as `max(f(v), 0)`.
+"""
+struct EpigraphViolationDistance <: AbstractDistance end
+
+"""
+    NormedEpigraphDistance{p}
+
+Distance used when `v ∈ s` can be expressed as a several functions `f_i(v) ≤ 0 ∀ i ∈ I`.
+The distance is expressed as `norm([max(f_i(v), 0) ∀ i ∈ I], n)` with `p` a norm value (2, 1, Inf).
```

Comment:
Could we use another norm value, apart of the usual ones `(2, 1, Inf)`?

Reply:
yes indeed, I rephrased it to indicate that any norm value that's legit for `LinearAlgebra.norm` can be accepted